src/commands/settings/leveling/rewards.ts

```diff
@@ -159,14 +159,16 @@ export default class Rewards {
 
   async getRewards(guildId: string): Promise<Reward[]> {
     const settingsTable = await getSettingsTable(this.db);
-    const rewards = new Promise((resolve, reject) => {
-      settingsTable?.get(`${guildId}.leveling.rewards`).then(rewards => {
-        if (!rewards) return resolve([]);
-        resolve(rewards);
-      }).catch(() => {
-        resolve([]);
-      });
+    const rewards = new Promise(resolve => {
+      settingsTable
+        ?.get(`${guildId}.leveling.rewards`)
+        .then(rewards => {
+          if (!rewards) return resolve([]);
+          resolve(rewards);
+        })
+        .catch(() => resolve([]));
     });
```

Comment:
```suggestion
    const rewards = new Promise(resolve => settingsTable?
        .get(`${guildId}.leveling.rewards`)
        .then(rewards => {
          if (!rewards) return resolve([]);
          resolve(rewards);
        })
        .catch(() => resolve([]));
    );
```
I'm not 100% sure, but if that works then noice _clicks_. (Well, it just removes something, it's just me being hyped for v0.1a1 I guess lol)

Reply:
Or we could make it even more wild by doing...
```suggestion
    const rewards = new Promise(resolve => settingsTable
        .get(`${guildId}.leveling.rewards`)
        .then(rewards => if (!rewards) return resolve([]))
        .then(resolve(rewards));
        })
        .catch(() => resolve([]));
    });
```
But that'll probably just make more sense and also I don't even know if it's allowed to have multiple `.then()`s in one statement to be honest. Probably not 😆

Reply:
ahaha i will look into it, thanks!

Reply:
next commit has it